### ISSUE TITLE
signature verification not working on Windows 10

### ISSUE THREAD

**Issue by DanielRuf** [phar-io/phive#163]
Not sure if this has to do with having gpg2 and gpg.
How I can I debug this?

```
$ ./phive.phar install https://phar.phpunit.de/phpunit-7.3.4.phar
Phive 0.11.0 - Copyright (C) 2015-2018 by Arne Blankerts, Sebastian Heuer and Contributors
Downloading https://phar.phpunit.de/phpunit-7.3.4.phar
[2KDownloading https://phar.phpunit.de/phpunit-7.3.4.phar.asc68 MB - 100%
[2KDownloading key 4AA394086372C20A
Trying hkps.pool.sks-keyservers.net (193.164.133.100)
Downloading https://hkps.pool.sks-keyservers.net/pks/lookup?search=0x4AA394086372C20A&op=index&options=mr
[2KSuccessfully downloaded key================>| 463 B / 463 B - 100%

        Fingerprint: D840 6D0D 8294 7747 2937 7831 4AA3 9408 6372 C20A

        Sebastian Bergmann <sebastian@php.net> (2014-07-19)
        Sebastian Bergmann <sebastian@thephp.cc> (2014-07-19)
        Sebastian Bergmann <sebastian@phpunit.de> (2014-07-19)
        Sebastian Bergmann <sb@sebastian-bergmann.de> (2014-07-19)
        Sebastian Bergmann <sebastian.bergmann@thephp.cc> (2014-07-19)
        Sebastian Bergmann <sebastian.bergmann@bbp-consulting.com> (2017-01-10)

        Created: 2014-07-19

Import this key? [y|N] y
[ERROR]   Signature could not be verified
```

**Comment by pxgamer** [comment]
I also had this for a while where `gpg4win` and `gpg` (in Git for Windows) were conflicting.

I had to make sure that mine was using `C:\Program Files\Git\usr\bin\gpg.exe` first in the path, but this is probably different. I think it ended up being something to do with permissions on my gpg configuration directory.

**Comment by DanielRuf** [comment]
Hm, can we set this as parameter for phive to tell it to use a specific gpg binary / path?

Because changing the path may have much more unwanted side effects (regarding keyrings and so on and clients who use the gpg / gpg2 binary).

**Comment by theseer** [comment]
Currently, no.

The path to the gnupg binary read via [Config::getGPGBinaryPath](https://github.com/phar-io/phive/blob/master/src/shared/config/Config.php#L70), which in turn uses [Environment::getPathToCommand](https://github.com/phar-io/phive/blob/master/src/shared/environment/Environment.php#L31) to determine the path.

On windows, the later uses `where.exe` to find it, which i presume would use the `PATH` environment variable.

tl;dr: We probably should add an option to do so.

**Comment by theseer** [comment]
On a related note: Maybe you could install `ext/gnupg` for PHP? That would liberate you from using an external gpg binary. `Phive` will detect and favor the installed extension so to only falls back to running the binary if it's not available.

I have no idea though whether or not that extension works on windows.

**Comment by DanielRuf** [comment]
> I'd be interested to try the `ext/gnupg` if it works. 👍

Not sure where this is available for Windows. I did not find any link for the dll downloads. Or are they shipped with Bitnami WAMP stack and others?

**Comment by mantis** [comment]
There's no build setup for the gnupg extension on windows at least, however spotting the existence of https://github.com/singpolyma/openpgp-php - would it be possible to do a php verification ?

**Comment by DanielRuf** [comment]
Closing as I do not use Windows for development anymore.
